pkg/manager/impl/util/single_task_execution.go

```diff
@@ -0,0 +1,340 @@
+	for key, val := range inputTemplate.Variables {
+		binding := &core.Binding{
+			Var: key,
+		}
+		var bindingData core.BindingData
+		if val.Type.GetSimple() != core.SimpleType_NONE {
+			if inputs.Literals[key] != nil {
+				bindingData = core.BindingData{
+					Value: &core.BindingData_Scalar{
+						Scalar: inputs.Literals[key].GetScalar(),
+					},
+				}
+			}
+
+		} else if val.Type.GetSchema() != nil {
+			if inputs.Literals[key] != nil && inputs.Literals[key].GetScalar() != nil {
+				bindingData = core.BindingData{
+					Value: &core.BindingData_Scalar{
+						Scalar: &core.Scalar{
+							Value: &core.Scalar_Schema{
+								Schema: inputs.Literals[key].GetScalar().GetSchema(),
+							},
+						},
+					},
+				}
+			}
+		} else if val.Type.GetCollectionType() != nil {
+			if inputs.Literals[key] != nil && inputs.Literals[key].GetCollection() != nil &&
+				inputs.Literals[key].GetCollection().GetLiterals() != nil {
+				collectionBindings := make([]*core.BindingData, len(inputs.Literals[key].GetCollection().GetLiterals()))
+				for idx, literal := range inputs.Literals[key].GetCollection().GetLiterals() {
+					collectionBindings[idx] = getBinding(literal)
+
+				}
+				bindingData = core.BindingData{
+					Value: &core.BindingData_Collection{
+						Collection: &core.BindingDataCollection{
+							Bindings: collectionBindings,
+						},
+					},
+				}
+			}
+		} else if val.Type.GetMapValueType() != nil {
+			if inputs.Literals[key] != nil && inputs.Literals[key].GetMap() != nil &&
+				inputs.Literals[key].GetMap().Literals != nil {
+				bindingDataMap := make(map[string]*core.BindingData)
+				for k, v := range inputs.Literals[key].GetMap().Literals {
+					bindingDataMap[k] = getBinding(v)
+				}
+
+				bindingData = core.BindingData{
+					Value: &core.BindingData_Map{
+						Map: &core.BindingDataMap{
+							Bindings: bindingDataMap,
+						},
+					},
+				}
+			}
+		} else if val.Type.GetBlob() != nil {
+			if inputs.Literals[key] != nil && inputs.Literals[key].GetScalar() != nil {
+				bindingData = core.BindingData{
+					Value: &core.BindingData_Scalar{
+						Scalar: &core.Scalar{
+							Value: &core.Scalar_Blob{
+								Blob: inputs.Literals[key].GetScalar().GetBlob(),
+							},
+						},
+					},
+				}
+			}
+		} else {
+			return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Unrecognized value type [%+v]", val.GetType())
+		}
+		binding.Binding = &bindingData
+		bindings = append(bindings, binding)
+	}
+	return bindings, nil
+}
+
+func CreateOrGetWorkflowModel(
+	ctx context.Context, request admin.ExecutionCreateRequest, db repositories.RepositoryInterface,
+	workflowManager interfaces.WorkflowInterface, namedEntityManager interfaces.NamedEntityInterface, taskIdentifier *core.Identifier,
+	task *admin.Task) (*models.Workflow, error) {
+	workflowModel, err := db.WorkflowRepo().Get(ctx, repositoryInterfaces.GetResourceInput{
+		Project: taskIdentifier.Project,
```

Comment:
We talked about creating these skeleton WFs in a dummy project/domain. So that we don't clutter the UI. Did that not work?

Reply:
unfortunately projects are registered dynamically although we could seed the database with a system project like we do in the sandbox but we also have https://github.com/lyft/flyteidl/blob/4c689bdb271db83c1934505282d5e72981f23a51/protos/flyteidl/admin/common.proto#L33 as a separate status denoting origin/status